closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -55,15 +150,189 @@ static void handleErrors(void) {
 
     printf("An error occurred\n");
 
-    while((errCode = ERR_get_error()) != 0)
+    while((errCode = (*OSSL_get_error)()) != 0)
     {
-        char *err = ERR_error_string(errCode, NULL);
-        printf("Generating error message\n" );
+        char *err = (*OSSL_error_string)(errCode, NULL);
         printf("%s\n", err);
     }
     abort();
 }
 
+/*
+ * Class:     jdk_crypto_jniprovider_NativeCrypto
+ * Method:    loadCrypto
+ * Signature: ()V
+ */
+JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+  (JNIEnv *env, jclass thisObj){
+
+    void *handle;
+    char *error;
+    typedef const char* OSSL_version_t(int);
+
+     // Determine the version of OpenSSL.
+    OSSL_version_t* OSSL_version;
+    const char * openssl_version;
+    int ossl_ver;
+
+    // Load OpenSSL Crypto library
+    handle = load_crypto_library();
+    if (handle == NULL) {
+        fprintf(stderr, " :FAILED TO LOAD OPENSSL CRYPTO LIBRARY\n");
```

Comment:
openssl is supposed to silently fail if the library cannot be loaded, and fall back to the java implementation. While this is fine for debugging, and we can have an option to turn on debug output, there shouldn't be anything printed to the console by default when loading the library fails.

Reply:
Yes, we have realised this. We were only using what Nasser had in his PR.